### PR TITLE
feat: SfBanner adding prop for button link

### LINKED ISSUE #1510: [FEATURE] SfBanner has no link prop and is not clickable on mobile
**Describe how this feature will help us improve**
1. There is no prop containing link for the CTA button in SfBanner.
2. In mobile mode when button disappears, whole SfBanner is not clickable. 

**Is your proposal related to a problem?**
1. When trying to pass the link to the CTA button you need to add the button inside slot, but would be quicker to pass the link inside the prop. 
2. After opening banner in mobile mode you can't click on it. It should behave like button. 

**Describe the solution you'd considered**
1. Add button-link prop to SfBanner and pass it to the call-to-action slot to SfButton.
2. Wrap whole banner into SfButton to make it clickable on mobile.

### PR DESCRIPTION
**Pull request by AdamPawlinski**
# Related issue
Closes #1510 

# Scope of work
Prop link added to SfBanner and button wrapper added in mobile mode to make whole banner clickable. 

# Screenshots of visual changes
image

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue [hunk 1]
```diff
@@ -1,6 +1,13 @@
 <template>
   <section class="sf-banner" :style="style" v-on="isMobile ? $listeners : {}">
-    <div class="sf-banner__container">
+    <SfButton
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfBanner/SfBanner.vue:3]: Now we have a button inside another button :)

**Review by justyna-13 — CHANGES_REQUESTED** [review]
See my comment :)

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue (hunk 1 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/molecules/SfBanner/SfBanner.vue]: I changed it to have a 'div' wrapper on desktop and 'a' or 'button' on mobile depending on passing the link prop or not. Wdyt @justyna-13 ?

File: packages/shared/styles/components/molecules/SfBanner.scss [hunk 2]
```diff
@@ -1,87 +1,69 @@
...
-    min-height: var(--banner-height, 25rem);
-    &__call-to-action {
-      --button-padding: var(--spacer-sm) var(--spacer-xl);
-    }
-    &--right {
-      --banner-justify-content: flex-end;
-    }
-  }
-}
+    @import "../../helpers";
```
Comment (justyna-13) [inline packages/shared/styles/components/molecules/SfBanner.scss:1]: indentation :)

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue [hunk 3]
```diff
@@ -19,17 +19,19 @@
       <slot name="call-to-action" v-bind="{ buttonText }">
         <SfButton
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfBanner/SfBanner.vue:20]: we still have our previous problem with btn in btn on mobile :) what do you think about adding an additional condition for showing this button only on desktop? :)

**Review by justyna-13 — COMMENTED** [review]
good job :) see my comments

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue (hunk 3 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/molecules/SfBanner/SfBanner.vue:20]: You are right, I add this condition.

File: packages/shared/styles/components/molecules/SfBanner.scss (hunk 2 above)
Comment (AdamPawlinski) [reply packages/shared/styles/components/molecules/SfBanner.scss]: fixed finally...

**Review by justyna-13 — DISMISSED** [review]

**Comment by AdamPawlinski** [comment]
@justyna-13 I made this 2 changes that you requested

**Review by justyna-13 — COMMENTED** [review]
one more thing - please update docs regarding this change :)

**Comment by AdamPawlinski** [comment]
I added info to releases docs.

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <template>
   <section class="sf-banner" :style="style" v-on="isMobile ? $listeners : {}">
-    <div class="sf-banner__container">
+    <component :is="tag" class="sf-banner__button-wrapper" :link="link">
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfBanner/SfBanner.vue:3]: if it's not only a button tag I'd change also its class name - what do you think about "wrapper" ? :)

**Review by justyna-13 — DISMISSED** [review]

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue (hunk 4 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/molecules/SfBanner/SfBanner.vue]: @justyna-13 fixed that, take a look please

File: packages/shared/styles/components/molecules/SfBanner.scss [hunk 5]
```diff
@@ -2,23 +2,30 @@
...
     var(--_banner-background-image)
   );
   --banner-background-position: 60%;
-  &__container {
+  &__wrapper {
     display: flex;
-    flex: 0 0 var(--banner-container-width, 70%);
+    flex-direction: column;
+    align-items: var(--banner-align-items, flex-start);
     justify-content: var(--banner-container-justify-content, flex-start);
```
Comment (justyna-13) [inline packages/shared/styles/components/molecules/SfBanner.scss:18]: would be good to change also custom properties names inside wrapper: ```--banner-container-justify-content``` to ```--banner-wrapper-justify-content``` :)

Comment (AdamPawlinski) [reply packages/shared/styles/components/molecules/SfBanner.scss:18]: changed

**Review by justyna-13 — APPROVED** [review]

**Pull request merged**